### PR TITLE
 Build: Rewrite test-on-node task to use QUnit.on()

### PR DESCRIPTION
**Pull request by Krinkle**
_Depends on: https://github.com/qunitjs/qunit/pull/1515_

-------

Ref #1511.

### CODE REVIEW

File: build/tasks/test-on-node.js [hunk 1]
```diff
@@ -5,63 +5,24 @@ module.exports = function( grunt ) {
...
-			todo: 0,
-			runtime: 0,
-			passedAssertions: 0,
-			failedAssertions: 0
-		} );
-
-		grunt.log.writeln( "-----" );
-		grunt.log.ok( constructMessage( totals ) );
+		if ( totals.failed ) {
+			grunt.log.error( `Ran ${totals.files} test files, ${totals.failed} failed` );
```
Comment (smcclure15) [inline build/tasks/test-on-node.js:15]: I always have to double-check: is "failed" the number of files failed, or the number of assertions failed across all tests/files?
If it's the latter, maybe rewording to "Ran X test files, Y assertions failed". Otherwise it reads fine, and it just my own hesitation.

**Review by smcclure15 — APPROVED** [review]
I like this - especially the cleaner/flatter logic inside src/test.js

File: build/tasks/test-on-node.js (hunk 1 above)
Comment (Krinkle) [reply build/tasks/test-on-node.js:15]: Neither assertions nor files, it's the number of tests.

Updated to:
```js
Ran ${totals.files} files, ${totals.failed} failed tests
```

**Comment by Krinkle** [comment]
- Fixed test counter.
- Merged https://github.com/qunitjs/qunit/pull/1515 and rebased on master.

**Comment by Krinkle** [comment]
I realized this task runs before the headless chrome and CLI tests, which provide greater debugging details. Which meant that in the case of failures in the "main" tests, the minimal test-on-node task was the first failure and thus prevented the others from running.

It is still the first one, but at least as of this commit it includes a stack trace. Once we've added the "main" tests to the CLI runs (ref https://github.com/qunitjs/qunit/issues/1511), we may want to re-arrange it so that the CLI and browser test run first.
